Helper/Order.php

```diff
@@ -1255,12 +1255,19 @@ public function deleteOrderByIncrementId($displayId)
 
     /**
      * @param $orderIncrementId
-     * @return OrderModel|false
+     * @param null $quoteId
+     * @return false|OrderInterface|OrderModel
      */
-    public function getExistingOrder($orderIncrementId)
+    public function getExistingOrder($orderIncrementId, $quoteId = null)
     {
         /** @var OrderModel $order */
-        return $this->cartHelper->getOrderByIncrementId($orderIncrementId, true);
+        $order = $this->cartHelper->getOrderByIncrementId($orderIncrementId, true);
+
+        if ($quoteId && (!$order || !$order->getId())) {
```

Comment:
Could you please add a small comment why we need this: like if order_id changed by 3p module and we had timeout issue

Reply:
@vitaliyreznikov I've updated. Can you re-check?